lib/resource_waiter.js

```diff
@@ -136,6 +136,16 @@ AWS.ResourceWaiter = inherit({
       callback = params; params = undefined;
     }
 
+    if (params && params.$waiter) {
+      if (params.$waiter.delay) {
+        this.config.delay = params.$waiter.delay;
+      }
+      if (params.$waiter.maxAttempts) {
+        this.config.maxAttempts = params.$waiter.maxAttempts;
+      }
+      delete params.$waiter;
```

Comment:
We typically try to treat user input as immutable, that way they can reuse the same object in another call if desired.